src/algorithms/signal_source/gnuradio_blocks/fifo_reader.cc

```diff
@@ -80,35 +88,102 @@ int FifoReader::work(int noutput_items,
     return this->WORK_CALLED_PRODUCE;
 }
 
+
+template <typename Type>
+size_t FifoReader::read_interleaved(int noutput_items, gr_vector_void_star &output_items)
+{
+    boost::mutex::scoped_lock lock(d_mutex);  // hold mutex for duration of this function
+    size_t items_retrieved = 0;
+    int n;
+    Type real;
+    Type imag;
+
+    if (buffer_idx_ >= buffer_size_)
+        {
+            fifo_.read(fifo_buffer_, FIFO_SIZE);
```

Comment:
I realize this is moved code, so your responsibility is limited. However, I'd make a point of noting that this will block until FIFO_SIZE is read completely. At least, I think that's the design.

Now, the question is, should you read more than `noutput_items`, since that's all you wanted?

Reply:
I do see the previous implementation only read `noutput_items` from the FIFO